include/EASTL/memory.h

```diff
@@ -1227,6 +1227,7 @@ namespace eastl
 	template <typename T>
 	inline void destruct(T* p)
 	{
+		(void)p;
```

Comment:
Please use our portable macro "EA_UNUSED" instead of casting to void.  Please reference the MSDN article in a comment saying this cast is necessary due to a limitation in MSVC.